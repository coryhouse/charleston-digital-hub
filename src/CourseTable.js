import React from "react";
import PropTypes from "prop-types";
import { course } from "./propTypes";
import { Link } from "react-router-dom";

const CourseTable = ({ courses, onClickDelete }) => {
  return (
    <table>
      <thead>
        <tr>
          <th />
          <th>Title</th>
          <th>Author</th>
          <th>Category</th>
        </tr>
      </thead>
      <tbody>
        {courses.map(course => (
          <tr key={course.id}>
            <td>
              <button onClick={event => onClickDelete(course.id)}>
                Delete
              </button>
            </td>
            <td>
              <Link to={"/course/" + course.slug}>{course.title}</Link>
            </td>
            <td>{course.authorId}</td>
            <td>{course.category}</td>
          </tr>
        ))}
      </tbody>
    </table>
  );
};

CourseTable.propTypes = {
  courses: PropTypes.arrayOf(course),
  onClickDelete: PropTypes.func.isRequired
};

CourseTable.defaultProps = {
  courses: []
};

export default CourseTable;
