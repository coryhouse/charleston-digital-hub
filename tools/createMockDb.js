// Create a db.json file using mockData.js as the source.
// This way json-server has consistent data to serve upon app start.
/* eslint-disable no-console */
const fs = require("fs");
const path = require("path");
const mockData = require("./mockData");

const data = JSON.stringify(mockData);
const filepath = path.join(__dirname, "db.json");

fs.writeFile(filepath, data, function(err) {
  err ? console.log(err) : console.log("Mock DB created.");
});
