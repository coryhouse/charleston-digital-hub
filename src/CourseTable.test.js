import React from "react";
import CourseTable from "./CourseTable";
import renderer from "react-test-renderer";
import { courses } from "../tools/mockData";
import { BrowserRouter as Router } from "react-router-dom";
import { render, cleanup } from "react-testing-library";

afterEach(cleanup);

describe("React testing library", () => {
  it("should render no courses message when no courses are passed", () => {
    // arrange
    const { getByText, debug } = render(
      <CourseTable courses={[]} onClickDelete={jest.fn()} />
    );

    // act
    // assert
    getByText("No courses.");
  });
});

it("should render courses when passed courses", () => {
  // arrange

  // act
  const tree = renderer.create(
    <Router>
      <CourseTable courses={courses} onClickDelete={jest.fn()} />
    </Router>
  );

  // assert
  expect(tree).toMatchSnapshot();
});

it("should render no courses message when passed no courses", () => {
  // arrange

  // act
  const tree = renderer.create(
    <Router>
      <CourseTable courses={[]} onClickDelete={jest.fn()} />
    </Router>
  );

  // assert
  expect(tree).toMatchSnapshot();
});
