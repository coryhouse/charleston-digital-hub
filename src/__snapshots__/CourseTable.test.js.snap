// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<table>
  <thead>
    <tr>
      <th />
      <th>
        Title
      </th>
      <th>
        Author
      </th>
      <th>
        Category
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        <button
          onClick={[Function]}
        >
          Delete
        </button>
      </td>
      <td>
        <a
          href="/course/react-auth0-authentication-security"
          onClick={[Function]}
        >
          Securing React Apps with Auth0
        </a>
      </td>
      <td>
        1
      </td>
      <td>
        JavaScript
      </td>
    </tr>
    <tr>
      <td>
        <button
          onClick={[Function]}
        >
          Delete
        </button>
      </td>
      <td>
        <a
          href="/course/react-big-picture"
          onClick={[Function]}
        >
          React: The Big Picture
        </a>
      </td>
      <td>
        1
      </td>
      <td>
        JavaScript
      </td>
    </tr>
    <tr>
      <td>
        <button
          onClick={[Function]}
        >
          Delete
        </button>
      </td>
      <td>
        <a
          href="/course/react-creating-reusable-components"
          onClick={[Function]}
        >
          Creating Reusable React Components
        </a>
      </td>
      <td>
        1
      </td>
      <td>
        JavaScript
      </td>
    </tr>
    <tr>
      <td>
        <button
          onClick={[Function]}
        >
          Delete
        </button>
      </td>
      <td>
        <a
          href="/course/javascript-development-environment"
          onClick={[Function]}
        >
          Building a JavaScript Development Environment
        </a>
      </td>
      <td>
        1
      </td>
      <td>
        JavaScript
      </td>
    </tr>
    <tr>
      <td>
        <button
          onClick={[Function]}
        >
          Delete
        </button>
      </td>
      <td>
        <a
          href="/course/react-redux-react-router-es6"
          onClick={[Function]}
        >
          Building Applications with React and Redux
        </a>
      </td>
      <td>
        1
      </td>
      <td>
        JavaScript
      </td>
    </tr>
    <tr>
      <td>
        <button
          onClick={[Function]}
        >
          Delete
        </button>
      </td>
      <td>
        <a
          href="/course/react-flux-building-applications"
          onClick={[Function]}
        >
          Building Applications in React and Flux
        </a>
      </td>
      <td>
        1
      </td>
      <td>
        JavaScript
      </td>
    </tr>
    <tr>
      <td>
        <button
          onClick={[Function]}
        >
          Delete
        </button>
      </td>
      <td>
        <a
          href="/course/writing-clean-code-humans"
          onClick={[Function]}
        >
          Clean Code: Writing Code for Humans
        </a>
      </td>
      <td>
        1
      </td>
      <td>
        Software Practices
      </td>
    </tr>
    <tr>
      <td>
        <button
          onClick={[Function]}
        >
          Delete
        </button>
      </td>
      <td>
        <a
          href="/course/architecting-applications-dotnet"
          onClick={[Function]}
        >
          Architecting Applications for the Real World
        </a>
      </td>
      <td>
        1
      </td>
      <td>
        Software Architecture
      </td>
    </tr>
    <tr>
      <td>
        <button
          onClick={[Function]}
        >
          Delete
        </button>
      </td>
      <td>
        <a
          href="/course/career-reboot-for-developer-mind"
          onClick={[Function]}
        >
          Becoming an Outlier: Reprogramming the Developer Mind
        </a>
      </td>
      <td>
        1
      </td>
      <td>
        Career
      </td>
    </tr>
    <tr>
      <td>
        <button
          onClick={[Function]}
        >
          Delete
        </button>
      </td>
      <td>
        <a
          href="/course/web-components-shadow-dom"
          onClick={[Function]}
        >
          Web Component Fundamentals
        </a>
      </td>
      <td>
        1
      </td>
      <td>
        HTML5
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`should render courses when passed courses 1`] = `
<table>
  <thead>
    <tr>
      <th />
      <th>
        Title
      </th>
      <th>
        Author
      </th>
      <th>
        Category
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        <button
          onClick={[Function]}
        >
          Delete
        </button>
      </td>
      <td>
        <a
          href="/course/react-auth0-authentication-security"
          onClick={[Function]}
        >
          Securing React Apps with Auth0
        </a>
      </td>
      <td>
        1
      </td>
      <td>
        JavaScript
      </td>
    </tr>
    <tr>
      <td>
        <button
          onClick={[Function]}
        >
          Delete
        </button>
      </td>
      <td>
        <a
          href="/course/react-big-picture"
          onClick={[Function]}
        >
          React: The Big Picture
        </a>
      </td>
      <td>
        1
      </td>
      <td>
        JavaScript
      </td>
    </tr>
    <tr>
      <td>
        <button
          onClick={[Function]}
        >
          Delete
        </button>
      </td>
      <td>
        <a
          href="/course/react-creating-reusable-components"
          onClick={[Function]}
        >
          Creating Reusable React Components
        </a>
      </td>
      <td>
        1
      </td>
      <td>
        JavaScript
      </td>
    </tr>
    <tr>
      <td>
        <button
          onClick={[Function]}
        >
          Delete
        </button>
      </td>
      <td>
        <a
          href="/course/javascript-development-environment"
          onClick={[Function]}
        >
          Building a JavaScript Development Environment
        </a>
      </td>
      <td>
        1
      </td>
      <td>
        JavaScript
      </td>
    </tr>
    <tr>
      <td>
        <button
          onClick={[Function]}
        >
          Delete
        </button>
      </td>
      <td>
        <a
          href="/course/react-redux-react-router-es6"
          onClick={[Function]}
        >
          Building Applications with React and Redux
        </a>
      </td>
      <td>
        1
      </td>
      <td>
        JavaScript
      </td>
    </tr>
    <tr>
      <td>
        <button
          onClick={[Function]}
        >
          Delete
        </button>
      </td>
      <td>
        <a
          href="/course/react-flux-building-applications"
          onClick={[Function]}
        >
          Building Applications in React and Flux
        </a>
      </td>
      <td>
        1
      </td>
      <td>
        JavaScript
      </td>
    </tr>
    <tr>
      <td>
        <button
          onClick={[Function]}
        >
          Delete
        </button>
      </td>
      <td>
        <a
          href="/course/writing-clean-code-humans"
          onClick={[Function]}
        >
          Clean Code: Writing Code for Humans
        </a>
      </td>
      <td>
        1
      </td>
      <td>
        Software Practices
      </td>
    </tr>
    <tr>
      <td>
        <button
          onClick={[Function]}
        >
          Delete
        </button>
      </td>
      <td>
        <a
          href="/course/architecting-applications-dotnet"
          onClick={[Function]}
        >
          Architecting Applications for the Real World
        </a>
      </td>
      <td>
        1
      </td>
      <td>
        Software Architecture
      </td>
    </tr>
    <tr>
      <td>
        <button
          onClick={[Function]}
        >
          Delete
        </button>
      </td>
      <td>
        <a
          href="/course/career-reboot-for-developer-mind"
          onClick={[Function]}
        >
          Becoming an Outlier: Reprogramming the Developer Mind
        </a>
      </td>
      <td>
        1
      </td>
      <td>
        Career
      </td>
    </tr>
    <tr>
      <td>
        <button
          onClick={[Function]}
        >
          Delete
        </button>
      </td>
      <td>
        <a
          href="/course/web-components-shadow-dom"
          onClick={[Function]}
        >
          Web Component Fundamentals
        </a>
      </td>
      <td>
        1
      </td>
      <td>
        HTML5
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`should render no courses message when passed no courses 1`] = `
<p>
  No courses.
</p>
`;
